src/TRestAxionOpticsMirror.cxx

```diff
@@ -0,0 +1,716 @@
+                    fSubstrate + "_" + fSigmaTop + ".N901f";
+    if (fMirrorType == "Bilayer")
+        fnameT = "Transmission_" + fMirrorType + "_" + fLayerTop + "_" + fLayerThicknessTop + "_" +
+                 fSigmaTop + "_" + fLayerBottom + "_" + fLayerThicknessBottom + "_" + fSigmaBottom + "_" +
+                 fSubstrate + ".N901f";
+    return fnameT;
+}
+
+///////////////////////////////////////////////
+/// \brief It is a private method to export the tables to a binary file once the tables
+/// have been downloaded from Henke database
+///
+Int_t TRestAxionOpticsMirror::ExportTables() {
+    if (fReflectivityTable.size() == 0) {
+        ferr << "Nothing to export!" << endl;
+        return 1;
+    }
+
+    string path = REST_USER_PATH + "/export/";
+
+    if (!TRestTools::fileExists(path)) {
+        cout << "Creating path: " << path << endl;
+        system(("mkdir -p " + path).c_str());
+    }
+
+    string fnameR = GetReflectivityFilename();
+    TRestTools::ExportBinaryTable(path + fnameR, fReflectivityTable);
+    info << "Reflectivity table generated at: " << path + fnameR << endl;
+
+    string fnameT = GetTransmissionFilename();
+    TRestTools::ExportBinaryTable(path + fnameT, fTransmissionTable);
+    info << "Transmission table generated at: " << path + fnameT << endl;
+
+    return 0;
+}
+
+///////////////////////////////////////////////
+/// \brief It downloads the reflectivity file for the present mirror properties
+/// defined at the metadata members.
+///
+/// \return It returns the location and filename of the downloaded file.
+///
+std::string TRestAxionOpticsMirror::DownloadHenkeFile() {
+    string perlName = "laymir.pl";
+    if (fMirrorType == "Bilayer") perlName = "bimir.pl";
+    string url = "https://henke.lbl.gov/cgi-bin/" + perlName;
+    string result = TRestTools::POSTRequest(url, fHenkeKeys);
+
+    size_t start = result.find("HREF=\"") + 6;
+    size_t length = result.find(".dat\">") + 4 - start;
+    result = result.substr(start, length);
+
+    return TRestTools::DownloadRemoteFile("https://henke.lbl.gov/" + result);
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the interpolated reflectivity for a given angle (in degrees)
+/// and a given energy (in keV).
+///
+Double_t TRestAxionOpticsMirror::GetReflectivity(const Double_t angle, const Double_t energy) {
+    if (fReflectivityTable.size() == 0) LoadTables();
+
+    Double_t en = energy;
+    if (en < 0.030) {
+        warning << "Energy is below 30eV! It should be between 30eV and 15keV" << endl;
+        warning << "Setting energy to 30eV" << endl;
+        en = 0.030;
+    }
+
+    if (en > 15) {
+        warning << "Energy is above 15keV! It should be between 30eV and 15keV" << endl;
+        warning << "Setting energy to 15keV" << endl;
+        en = 15;
+    }
+
+    Int_t lowEnBin = (Int_t)((en - 0.03) / 0.03);
```

Comment:
This cast from double to int always truncates the double. Is that your intended behavior? If so I'd add a comment stating as such.

Reply:
Yes, it is the intended behaviour. So, what do I have to write in the comment? That it is the intended behaviour? The cast is there for a reason, it is intentional. I am not in favor to add excessive comments. The code should talk by itself. It is retrieving the bin from a histogram correspoding to a given energy. Of course, there are other ways to do it, but for me this one is kind of intuitive.

Thats why I just resolved this comment without reply, dont think is necessary to address this.